\chapter{Results}\label{chap:results}
The Galaxy workflows are validated using real-world datasets from different laboratories. The analysis results for each workflow with complying test samples are described below.

\section{Validation of Poxvirus Workflow on Lumpy Skin Disease Virus Datasets}
We employed our pipeline using a tiling amplicon approach with masked reference sequences for each half genome. Two public \ac{LSDV} samples, 20L70 (SRR15145276 \todo{oder accession mit SRX angeben?} and SRR15145275) and 20L81 (SRR15145274 and SRR15145273), that were sequenced with the required tiled amplicon method in two pools are used. Collected from cattle in 2020 during a lumpy skin disease outbreak in Nothern Vietnam (20L70\_Dinh-To/VNM/20 and 20L81\_Bang-Thanh/VNM/20), both samples were sequenced on an MiSeq System using a Nextera XT DNA library preparation kit (Illumina). The used \acs{CaPV} primer scheme in \ac{BED} format contains pool identifier in the \textit{SCORE} column. The widely-used vaccine ``Neethling'' strain was used as reference genome (NC\_003027). \\
The raw FASTQ files for each sample were trimmed with \texttt{fastp} so that over 

\setlength{\tabcolsep}{14pt}
\renewcommand{\arraystretch}{1.3}
\begin{table}[ht!]
    \centering
    \begin{tabular}{lcc}
    \toprule
    \textbf{Output Metric}                                                              & \textbf{20L70}  & \textbf{20L81} \\ \midrule
    Paired-end raw reads                                                                  & 863,820        & 1,016,168      \\ \hdashline
    \begin{tabular}[c]{@{}l@{}}Paired-end reads after\\ quality trimming\end{tabular}     & 856,128        & 947,064        \\ \midrule
    \begin{tabular}[c]{@{}l@{}}Proportion of reads\\ mapping to NI-2490 (\%)\end{tabular} & 99.6           & 77.3           \\ \hdashline
    \begin{tabular}[c]{@{}l@{}}Proportion of NI-2490\\ genome covered\end{tabular}        & 100\%          & 100\%          \\ \bottomrule
    \end{tabular}
    \caption{Summary of the main metrics for datasets 20L70 and 20L81.}
    \label{tab:pox-metrics}
\end{table}

\todoit
\ac{IWC} link, primer scheme.

%NC\_003027.1 as reference sequence (South African Neethling strain. Collection date: 1959 from Bos taurus)
-> also used for mostly used vaccine. Other strains are wildtype LSDV and KSGP strains (Accession: PRJNA661421, SRA: SRS7321935)


\section{Validation of AIV Workflow on H4N6 and H5N8 Samples}\label{sec:4-aiv}
\todoit

We illustrate the utility of the workflow on the Galaxy platform, we use public real-world datasets

Reference database for each influenza gene segment, prepared the collection from public INSaFLU/NCBI data. Consists of XX many sequences

within-subtype variation IS captured well by this ref. database (many references per subtype)

%(not) controlling influenza A/B species assignment: 50 informative references

Samples by Sciensano s4+s8
% U2008751-n5_S4 (H4N6)
% U2012100-n21_S8 (H5N8)

point out output for downstream analyses 

Quality report, snipit plots, IQ-Tree for \ac{HA}/\ac{NA}, consensus reference, VAPOR scores

\section{Validation of FMDV Workflow on ? Samples}
O/SAT2 samples
\todoit
Samples by Pirbright Institute by Dr. Graham Freimanis (NOT YET)

...wait for samples

\section{Workflow Profiling}
\todoit
Assembly vs. mapping
 