\chapter{Results}\label{chap:results}
The Galaxy workflows are validated using real-world datasets from different laboratories. The analysis results for each workflow with complying test samples are described below.

\section{Validation of Poxvirus Workflow on Lumpy Skin Disease Virus Datasets}
We employed our pipeline using a tiling amplicon approach with masked reference sequences for each half genome. Two public \ac{LSDV} samples, 20L70 (SRR15145276 \todo{oder accession mit SRX angeben?} and SRR15145275) and 20L81 (SRR15145274 and SRR15145273), that were sequenced with the required tiled amplicon method in two pools are used. Collected from cattle in 2020 during a lumpy skin disease outbreak in Northern Vietnam (20L70\_Dinh-To/VNM/20 and 20L81\_Bang-Thanh/VNM/20), both samples were sequenced on an MiSeq System using a Nextera XT DNA library preparation kit (Illumina). The used \acs{CaPV} primer scheme in \ac{BED} format contains pool identifier in the \textit{SCORE} column. The widely-used vaccine ``Neethling'' strain was used as reference genome (NC\_003027). \\
The raw FASTQ files for each sample were trimmed with \texttt{fastp} and mapped to each reference half. \\

\setlength{\tabcolsep}{16pt}
\renewcommand{\arraystretch}{1.3}
\begin{table}[ht!]
    \centering
    \begin{tabular}{lcc}
    \toprule
    \textbf{Output Metric}                      & \textbf{20L70}     & \textbf{20L81}     \\ \midrule
    Paired-end raw reads                        & 863 820            & 1 016 168          \\ 
    Paired-end reads after quality trimming     & 856 128            & 947 064            \\ \midrule
    Proportion of reads mapping to NI-2490      & 99.6\%             & 77.3\%             \\ 
    Proportion of NI-2490 genome covered        & 100\%              & 100\%              \\ \midrule
    Mean coverage                               & 2 705.2 \texttimes & 2 411.4 \texttimes \\ 
    Alignment error rate                        & 1.25\%             & 1.30\%             \\ \bottomrule
    \end{tabular}
    \caption{Metrics after preprocessing and mapping for datasets 20L70 and 20L81.}
    \label{tab:pox-metrics}
\end{table}

The used primer scheme contains a total of 23 primers, while the first 12 primers cover \textit{pool1} and the remaining 11 primers cover \textit{pool2}. Inspection of the masking intervals for N-masking the reference confirms that the right-most position of Ns of masking the first half (i.e. preparing the reference for mapping \textit{pool2} reads) is the first start position of \textit{pool2} primers (``1 -- 79081'', and 79081 being the start position of primer 13). Accordingly for N-masking the reference for mapping of \textit{pool1} reads, the interval from the right-most primer-end of \textit{pool1} primers is 80202, the end position of primer 12, resulting in the interval ``80202 -- 150773''. The final position is the remainder of the reference sequence. Since the reference genome and primer scheme are the same for both datasets 20L70 and 20L81, the N-masked reference is used for both mappings. Mapping of each pool is done with \texttt{BWA-MEM} and default settings for Illumina-sequenced reads, using the N-masked reference for \textit{pool1} and \textit{pool2} respectively. After merging the mappings with \texttt{Samtools merge}, statistics for preprocessing and mapping are reported and summarised in~\tabref{tab:pox-metrics}. \\
The merged mapping of both read pools is quality trimmed with \texttt{iVar trim} to remove primers and reads with a length of less than 30. The remaining reads are used for full-length consensus sequence construction with \textit{iVar consensus}, developed for amplicon-based sequencing data. Inspecting the consensus sequences for both samples, %%%

\todoit
\ac{IWC} link, primer scheme?

%NC\_003027.1 as reference sequence (South African Neethling strain. Collection date: 1959 from Bos taurus)
-> also used for mostly used vaccine. Other strains are wildtype LSDV and KSGP strains (Accession: PRJNA661421, SRA: SRS7321935)


\section{Validation of AIV Workflow on H4N6 and H5N8 Samples}\label{sec:4-aiv}
\todoit

We illustrate the utility of the workflow on the Galaxy platform, we use public real-world datasets

Reference database for each influenza gene segment, prepared the collection from public INSaFLU/NCBI data. Consists of XX many sequences

within-subtype variation IS captured well by this ref. database (many references per subtype)

%(not) controlling influenza A/B species assignment: 50 informative references

Samples by Sciensano s4+s8
% U2008751-n5_S4 (H4N6)
% U2012100-n21_S8 (H5N8)

point out output for downstream analyses 

Quality report, snipit plots, IQ-Tree for \ac{HA}/\ac{NA}, consensus reference, VAPOR scores

\section{Validation of FMDV Workflow on ? Samples}
O/SAT2 samples
\todoit
Samples by Pirbright Institute by Dr. Graham Freimanis (NOT YET)

...wait for samples

\section{Workflow Profiling}
\todoit
Assembly vs. mapping
 