\chapter{Discussion}\label{chap:discussion}
\todoit

\section{Contribution to the Field}
\todoit
Workflows that solve common problems, provide useful information, are user-friendly, customisable, extendable

single sample vs. multi sample (reality check, what is needed?) \\
further pox viruses, pipelines can be more or less easily applied/adjusted \\
limitations \\
\ac{LSDV} interesting for all poxviruses and adjustable due to \acp{ITR} and tiled-amplicon approach \\
\ac{AIV} downstream - everything is possible. Highlight key minor assets that indicate adaption to mammals -> databases needed to check against, detect mutation of isolates? \\
Generally: annotate on amino acid layer (most information) \\

AIV: Choice of reference, why not BLAST or lookup in GISAID with many hundereds of thousands of RNA virus genomes; \\
AIV: Alternatives, such as read classification by mapping to a large database of influenza sequences and subsequent de novo assembly can help to resolve this issue, such pipelines are often complex, slow, and require expertise that is not necessarily available in routine
surveillance or public health laboratories. Secondly, even if bioinformatics pipelines are chosen judiciously, sequences of zoonotic origin may fail to be identified, resulting in a dataset that appears to below coverage, missing segments, or missing potential future pandemic reassortments. Furthermore, even with recent assembly programs, misassembly can occur.

However using an optimal reference is ideal, since for later advanced applications, such as transmission events, or study of intra-host variation, the closest possible reference may be necessary. Whilst BLAST performed well for individual read classification, it is often too slow for general application. With regards to de novo assembly, in the cases where not enough initial data exists to assemble fragments, mapping allows analysis of limited fragments. Furthermore, assembly of virus genomes can be slow, often taking several days for a single sample when contaminant reads are present.

Make phylogenetic trees publicly accessible, not one sample per strain but in high resolution and greater details, strains from different countries,

``The high sensitivity of the \ac{NGS} technology ensures that major kinds of viral pathogens in mixed samples can be detected.''
One strength of \ac{NGS} is that it can be used to detect emerging viral diseases with a high genetic variation. Like \ac{AIV}. Since it can analyse a full sequence instead of targeting a specific gene. -> makes sense to use virus-specific primers for \ac{PCR} or \ac{NGS} 

``Comparison of the whole genome sequences of recent \ac{LSDV} isolates from the 2015â€“2016 epidemic in southern Europe revealed only a limited number of point mutations between the isolates'' \ac{WGS} is essential to capture all genetic variation at once

In sequencers, false positive variants (\ac{FPV}) must be avoided (happens when too many amplification cycles are made)

BWA-MEM vs. BWA-MEM2 for small viral genomes: no big speed gain, error-prone as a new tool. preferably stick to known tools in Galaxy that are proved to work well and have a maintenance

-> caveat of tools with small development teams: abundance of technological advancements or error solving, but dependence on the reliability -> choose established software

% Efficiency: Assembly vs. Mapping; efficiency
% Kraken2 vs. VAPOR; 
% Efficiency: LoFreq vs. iVar consensus; both consensus identification methods using the same site-specific depth threshold

% If the goal in a broad and rapid surveillance is a high number of sample throughput and analysis, assembly is too cost and time sensitive. The presented pipeline could be used in a broad context for the use in many laboratories.
% building index is expensive (BWT)

\section{Future Directions}
\todoit
further validation and improvement of the developed pipelines, expansion to other viral livestock diseases, integration with existing surveillance systems; expand the \ac{VETLAB} network to entitle even more professionals to professionally analyse their samples.

\ac{AIV} workflow offers many possible directions for downstream analysis:

* consensus sequence for each segment -> compare consensus sequence to others can help identify outbreaks and patterns of transmission, get more insights how the virus spreads and its evolution
* Prokka annotation file. Predict the protein coding regions of the virus, to understand the function of the viral proteins and how they interact with host cells
* SNPs relative to the reference sequence
* \ac{MSA} and phylogenetic tree for broad or detailed phylogenetic analysis and understand evolutionary relationships between the sample and other strains. could also use clusters or subtypes within the sample. make trees available so that new isolates can be immediately arranged
* more visualisation of the data

* long-term objective: build public high-resolution databases to enable researchers to detect mutation of an isolate. this is crucial for a global surveillance system to work.

* develop workflows for the same purpose that work with other NGS sequencing data (ONT, PacBio etc.)