\chapter{Discussion and Outlook}\label{chap:discussion}
Despite enormous efforts in the prevention and containment of viral livestock diseases, viruses are a constant threat to the economy and the livestock industry. Extensive information on viruses is being collected and shared at the international level, and research advances in diagnostics, epidemiology and virology are increasingly providing valuable insights into the genomic characteristics of specific viruses. Sequence analysis plays an important role in identifying outbreak sources and monitoring antigenic changes within known virus strains. Continuous surveillance, vaccine development and quality control are important aspects of global disease control programs to protect animals, livestock industry and public health. Therefore, a public infrastructure for modern surveillance methods for the use of \ac{NGS} is of global interest. Reconstruction of the full-length viral genome at the base level from raw sequencing reads requires practical knowledge in tool selection and parameter settings. However, \ac{WGS} is valuable for virus surveillance as it provides comprehensive genetic information about the virus sample. This information is used to identify virus strains and disease-causing mutations, detect within-host variation and determine relationships between populations and individual samples. Moreover, research on \ac{SARS-CoV-2} which was used to combat the \ac{COVID-19} pandemic, has made such great strides in understanding viral structures that the introduction of a public infrastructure for experts and laypeople suggests that this newfound knowledge can be applied to other emerging viruses. Existing pipelines for full genomic sequence analysis from raw read data are specific to a small subset of viruses, require domain expertise and a server infrastructure to run computations on local computers. Different sequencing platforms produce different raw read data types with specific needs. As part of the \ac{ZODIAC} project, we have developed three ready-to-use, publicly available workflows on the Galaxy platform for sequence analysis of poxviruses, \ac{AIV} and \ac{FMDV} from Illumina-sequenced reads to analyse these viral diseases in livestock by generating the full-length consensus sequence. With accompanying training resources and user-friendly workflows on an open-source platform, we ultimately contribute to research-based surveillance methods and enable health professionals without in-house resources to analyse viral outbreaks in livestock.\\
In this chapter, the contributions to the field, limitations of the work and an outlook for future directions with the workflows are discussed.

Aiming at the challenging whole-genome construction from raw sequencing data, we developed three Galaxy workflows that work with relevant livestock viruses and apply reference-based mapping approaches to determine the sample-specific full-length genome. By carefully choosing reference sequences for mapping, it is possible to avoid \textit{de novo} assembly and to integrate fast reference-based mapping tools. Our workflows for poxviruses, avian influenza virus and foot-and-mouth disease virus are all set up for usage with little to no knowledge required from the user. \ac{NGS} data for all three workflows are required to be produced from an Illumina platform which is one commonly used sequencing technology. We maintain the workflows on the community-based Galaxy platform that aims at providing a transparent, public infrastructure research platform for biomedical applications. Since Galaxy is a globally known platform, it is a reasonable choice for sharing the workflows and making them accessible to researchers with different backgrounds all over the world. The workflow \textit{.ga} format is convertable to other standard pipeline formats (e.g. \ac{CWL}) and can be exported for usage in other environments than Galaxy. \\
Our workflows are designed to streamline the process of whole-genome construction and reduce the time and effort required for this complex task. By leveraging the power of reference-based mapping, the computational challenges of \textit{de novo} assembly are avoided and accurate results are achieved more quickly. Comparing average run times and memory usage of a virus-specific assembler, \texttt{rnaviralSPAdes}, mapping with a standard alignment tool like \texttt{BWA-MEM} outperforms assembly with a small viral genome in all means. However, when it comes to larger genomes, assembly can become significantly slower and more resource-intensive process. Reference-based mapping remains a fast and efficient option even in such cases.\\
The methods used in the developed workflows take into account the accuracy, speed and usability of the choice of alignment and consider virus-specific characteristics throughout the entire workflow.\\
Adaption of curated Galaxy workflows that were designed for sequence analysis of \ac{SARS-CoV-2} samples allows to use tested processes and components within all newly developed workflows. This includes preprocessing, reference-based mapping, quality filtering and consensus sequence construction. Advantages of reusing components of the \ac{SARS-CoV-2} workflows are that they have been optimised for specific analysis tasks and have been exhaustively tested with real-world samples by the community. This encourages to leverage the expertise of others and avoid potential pitfalls and errors in the analysis.

Especially with the poxvirus workflow that not only performs fast for the large genome, but also succeeds to map the repeated region of identical \acp{ITR} at each genome end, a workflow with many applications has been developed. To avoid ambiguous mapping of the reads to either one of the \acp{ITR}, our workflow employs a tiling mapping approach from ampliconic sequencing data that are required to be sequenced in two pools. This method assures the correct read mapping throughout the whole genome and works for all poxviruses. The workflow requires raw Illumina-sequenced reads, an annotated primer scheme in \ac{BED} format that contains information about the sequencing pools, and a reference sequence for mapping. The user has the option to change default thresholds for the consensus calling step by configuring values for the minimum quality score to call base, the allele frequency threshold to call \acp{SNV} and the allele frequency threshold to call a consensus indel. The two \ac{LSDV} test samples which were used to assess the workflow quality yield considerable results and complete consensus sequences for each sample, however more and diverse test samples would be needed to determine the ability of the workflow to construct the consensus sequence from raw reads. \\
\todo{add results/discussion about lsdv/capv experiment. which ref. to choose -> which clade, pre-workflow with vapor for sub-sequence of meaningful positions to determine best ref. among capvs}
%  We showed that for Capripoxviruses, by choosing a different reference sequence from another \ac{CaPV} member for mapping, the final consensus sequence does not diverge from the other consensus genomes built based on a reference sequence from the same species. Moreover, the low mutation rate of poxviruses and few variations among samples suggest that a fixed choice of curated reference sequence is judicious. -- true? \\
% \todo{to find meaning of SNPs: take gff file for pox virus and identify changes on amino acid level in each ORF with SnpEff. Other option: check alignment of amino acids to identify mutation in the sample in case of deviation}
The poxvirus workflow is published and available for download on WorkflowHub and Dockstore, two popular and widely used databases for versioned pipelines. Additional training material and the used primer scheme for Capripoxviruses are linked and provided alongside the workflow, so that a user can be guided through the steps for a deeper understanding of the single workflow steps. Despite the specific amplicon-based sequencing in two pools is no generally applied method in the real world, it can be implemented in laboratories that work with the given workflow and facilitates the read data in the required formats. However in future version of this workflow, alternative sequencing platforms and simplified primer schemes can be considered to make more analyses possible from other raw sequencing data. Especially for Capripoxviruses, a new workflow prior to the presented workflow could help determining the reference sequence that should be used with the sample. In this pre-workflow, the genomic positions that differentiate the \ac{LSDV} strains and the \ac{SPPV} and \ac{GTPV} from each other, could be extracted to find the best fitting reference sequence with a \texttt{VAPOR} run against a \ac{CaPV} sequences database. Using this approach, similar to the \ac{FMDV} workflows, the user would be asked to infer the most similar sequence from the database search, however it ensures a high-quality alignment and low-error consensus sequence. \\
A similar challenge in sequence analysis of the avian influenza virus lays in finding the ideal reference sequence for mapping. Assembling raw reads may result in a complete genome too, however is more computationally intensive respecting the size of the genome and may end in misassembly regions where coverage criteria are not met. In the presented \ac{AIV} workflow, we use a new approach that compiles a hybrid reference from a large database that contains thousands of sequences for each of the eight \ac{AIV} segments. The user of the workflow is not required to enter an arbitrarily selected reference genome, instead the workflow automatically finds the most similar sequence from the database by using the search classification tool \texttt{VAPOR} and stacks its results together to one reference sequence. While it works fast on a large number of input reads and a query database consisting of thousands of sequences, it selects the highest scoring sequence from the database per segment based on a scoring function on a weighted De Bruijn graph. This method is less biased than mapping raw reads to an assembled contig and yields reliable results with high identities between the query reads and the found sequence for each of the segments in all test samples. Alternative methods to find similar sequences to use as reference, such as read classification by looking up large databases of full influenza genomes are often complex, slow on the large number of reads and require expertise from the user which is not necessarily available. Kraken2 as a taxonomic classification system, also based on k-mer matches similar to \texttt{VAPOR}, provides large databases of viral sequences. However, this tool requires more maintenance and computational resources than \texttt{VAPOR}, which was specifically developed for influenza reads. Using Kraken2 on one of the large Galaxy server instances comes with the caveat of older database versions that are maintained from within Galaxy and need regular updates, whereas the reference collection that \texttt{VAPOR} is based on can be easily maintained and expanded by the user. The overall good quality in the H4N6 and H5N8 test samples emphasises that this method finds the same consensus sequence as the assembly of the reads. These results encourage to test and use the workflow with more \ac{AIV} samples. The criteria for sequences within the reference genome database assure that complete proteins are captures within each gene by only retaining sequences that contain the complete start and stop codons and no ambiguous nucleotides. Moreover, the amount of sequences per subtype guarantees to capture within-subtype variation which is specifically vital for the detection and identification of nucleotide sites that differ from known strains. The reference collection can be extended with custom references by the user and therefore permits outbreak-specific analysis and taxonomic classification while searching closely related sequences. For investigation on base level on an \ac{AIV} genome, possible adaptations or reassortment events are captured and provide a useful starting point for further downstream analyses. The developed workflow provides multiple datapoints for examination within or outside Galaxy, and includes a summary of \acp{SNP} on gene level. Phylogenetic classification with \texttt{IQ-Tree} based on sequences from the reference database or extended with other samples outside the collection is a part of the workflow to allow insights into relations and clusters among the samples. In future versions of the \ac{AIV} workflow, this step could be exchanged with the \texttt{UShER}, a faster tree generation tool. However, the tool is not available on the Galaxy EU server instance in its high performant version. Additionally, for lineage classification, tools like Pangolin that are \ac{SARS-CoV-2} specific could be extended in the future to use it with other viruses. Outbreak tracing and analyses to determine transmission events, intra-host variation and virus origins can be started from these data. Other possible analyses include variant calling, lineage assignment, gene annotation, functional analysis and many more. These opportunities for downstream analysis are more exhaustive than other existing pipelines for genomic sequence analysis of \ac{AIV} samples, and while subtype identification remains an essential part of surveillance methods in the field, modern monitoring networks for zoonotic animal diseases require detailed study of the sequenced samples on the full genome. Although the workflow design is different to the underlying concept of the \ac{SARS-CoV-2} workflow, it picks up its major components while considering the \ac{AIV} specific viral attributes. Yet there is no integration of variant calling and gene annotation tools, which should be part of the workflow for meaningful analysis of the results. Also, the consensus calling tool \texttt{iVar consensus} as opposed to \texttt{LoFreq} which is used in the \ac{SARS-CoV-2} workflow, is a faster alternative that identifies the most frequent base at each position. Since this first version of the workflow ends by constructing the consensus sequence and marks some further directions for biomedical analysis, depending on research objectives and routine applications it could be expanded in the future. 

% \ac{FMDV} is a highly contagious virus and an economically devastating disease in cloven-hoofed animals such as cattle, pigs, sheep, goats, and deer. Therefore, advances in the surveillance to track transmission routes and contain outbreak consequences are of high importance for the modern livestock industry. A high number of sample throughput that undergoes analysis on genomic level requires capable and reliable infrastructures.

For large viral genomes, the generation of the genome sequence from \ac{NGS} reads by assembly is very cost and time sensitive. However, small viral genomes such as \ac{FMDV} perform reasonably faster in a \textit{de novo} assembly than larger \ac{DNA} genomes and achieve sufficiently good results. We therefore developed a Galaxy workflow that provides the necessary steps in two parts which first assembles long contigs from the raw Illumina-sequenced reads and searches the \ac{BLAST}n database for similar sequences, and secondly builds the consensus sequence from a reference-based alignment of the reads. Although the \ac{BLAST}n database that is used within Galaxy requires regular updates and is not always up-to-date compared to the \ac{BLAST} web form provided by \ac{NCBI}, it nevertheless obtains highly similar sequences from its database to proceed with. However, strains from more recent outbreaks are not guaranteed to be included in the database builds, and results may miss better matches and exclude results from recently examined samples. After the \ac{BLAST}n search, the user is asked to select a plausible reference sequence from the results, with which the second workflow for mapping and consensus generation can be started. Users without expertise in the field are guided to select a reference from the result by accompanying materials. The trade-off for integrating a \textit{de novo} assembly to determine a suitable reference sequence and achieving a high quality consensus sequence, but investing the time and computational resources in the assembly of the short viral genome is a convenient approach for genomic analysis of \ac{FMDV} samples. A characteristic of the \ac{FMDV} genome and of other Picornaviruses is its polyC tract in the 5' end. It is highly conserved across serotypes and both for mapping and assembly approaches poses a challenge to unambiguously map the reads in this location. In the four test samples used for workflow validation, the coverage decreased in these regions and a consensus base could not be called at each position. Despite this caveat, the workflow finds a reference sequence according to the sample serotype, reliably maps the reads to the reference and provides opportunities to adapt tool settings such as coverage thresholds to account for a low number of reads.\\
The high quality of consensus sequences obtained in the test samples show that the workflow is capable of providing genomic information on base resolution that allows insights into phylogenetic analysis and genomic surveillance. With intermediate steps that split the workflow in two parts, the user can early on detect possible contamination and co-infection in the sample. This insight leverages the workflow to be a valuable tool to detect genetic variations, mutations and viral characterisation for \ac{FMDV} samples.

The presented workflows for poxviruses, \ac{AIV} and \ac{FMDV} are embedded on the Galaxy platform and provide transparent analysis tools for genomic research on viral samples. They contribute to modern surveillance pipelines in the biomedical field by using fast, accurate reference-based mapping methods to generate consensus sequences that require no user expertise and prior knowledge about the sequenced samples. Despite more validation and improvements should be employed on the workflows, they help the 

