\chapter{Discussion}\label{chap:discussion}
%%% rephrase problems/questions
Despite enormous efforts in the prevention and containment of viral livestock diseases, viruses are a constant threat to the economy and the livestock industry. Extensive information on viruses is being collected and shared at the international level, and research advances in diagnostics, epidemiology and virology are increasingly providing valuable insights into the genomic characteristics of specific viruses. Sequence analysis plays an important role in identifying outbreak sources and monitoring antigenic changes within known virus strains. Continuous surveillance, vaccine development and quality control are important aspects of global disease control programs to protect animals, livestock industry and public health. Therefore, a public infrastructure for modern surveillance methods for the use of \ac{NGS} is of global interest. Reconstruction of the full-length viral genome at the base level from raw sequencing reads requires practical knowledge in tool selection and parameter settings. However, \ac{WGS} is valuable for virus surveillance as it provides comprehensive genetic information about the virus sample. This information is used to identify virus strains and disease-causing mutations, detect within-host variation and determine relationships between populations and individual samples. Moreover, research on \ac{SARS-CoV-2} which was used to combat the \ac{COVID-19} pandemic, has made such great strides in understanding viral structures that the introduction of a public infrastructure for experts and laypeople suggests that this newfound knowledge can be applied to other emerging viruses. Existing pipelines for full genomic sequence analysis from raw read data are specific to a small subset of viruses, require domain expertise and a server infrastructure to run computations on local computers. Different sequencing platforms produce different raw read data types with specific needs. As part of the \ac{ZODIAC} project, we have developed three ready-to-use, publicly available workflows on the Galaxy platform for sequence analysis of poxviruses, \ac{AIV} and \ac{FMDV} from Illumina-sequenced reads to analyse emerging viral diseases in livestock by generating the full-length consensus sequence. With accompanying training resources and user-friendly workflows on an open-source platform, we ultimately contribute to research-based surveillance methods and enable health professionals without in-house resources to analyse viral outbreaks in livestock.

\section{Contribution to the Field}
%%% Summarise key findings: both the significant and non-significant results
Aiming at the challenging whole-genome construction from raw sequencing data, we developed three Galaxy workflows that work with relevant livestock viruses. By carefully choosing reference sequences, it is possible to avoid \textit{de novo} assembly and integrate a fast reference-based mapping tool. Our workflows for poxviruses, avian influenza virus and foot-and-mouth disease virus are all set up for usage with little to no knowledge required from the user. NGS data for all three workflows are required to be produced from an Illumina platform which is one commonly sequencing technology. We maintain the workflows on the community-based Galaxy platform that aims at providing a transparent, public infrastructure research platform for biomedical applications. Since Galaxy is a globally known platform, it is a reasonable choice for sharing the workflows and making them accessible to researchers with different backgrounds all over the world.\\
Our workflows are designed to streamline the process of whole-genome construction and reduce the time and effort required for this complex task. By leveraging the power of reference-based mapping, the computational challenges of \textit{de novo} assembly are avoided and accurate results are achieved more quickly. Comparing average run times and memory usage of a virus-specific assembler, \texttt{rnaviralSPAdes}, mapping with a standard alignment tool like \texttt{BWA-MEM} outperforms assembly with a small viral genome in all means. However, when it comes to larger genomes, assembly can become significantly slower and more resource-intensive process. Reference-based mapping remains a fast and efficient option even in such cases.\\
The methods used in the developed workflows take into account the accuracy, speed and usability of the choice of alignment and consider virus-specific characteristics throughout the entire workflow.\\
Adaption of curated Galaxy workflows that were designed for sequence analysis of \ac{SARS-CoV-2} samples allowed to use tested processes and components within the workflow. Advantages of reusing components of these workflows are that they have been optimised for specific analysis tasks and have been exhaustively tested with real-world samples by the community. This encourages to leverage the expertise of others and avoid potential pitfalls and errors in the analysis.

\subsubsection*{Poxvirus Sequence Analysis}
Poxviruses as large, double-stranded virus \ac{DNA} genomes are among the largest known viruses to infect humans and animals. Due to their size, sequencing and assembly of the poxvirus genome can be challenging and traditional \textit{de novo} assembly approaches regularly fail. Therefore, we use a reference-based assembly method in our poxvirus workflow that not only performs fast for the large genome, but also tackles the repeated region of identical \acp{ITR} at each genome end. To avoid ambiguous mapping of the reads to either one of the \acp{ITR}, our workflow employs a tiling mapping approach from ampliconic sequencing data that are required to be sequenced in two pools. This method assures the correct read mapping throughout the whole genome and works for all poxviruses. The workflow requires raw Illumina-sequenced reads, an annotated primer scheme in \ac{BED} format that contains information about the sequencing pools, and a reference sequence for mapping. The user has the option to change default thresholds for the consensus calling step by configuring values for the minimum quality score to call base, the allele frequency threshold to call \ac{SNV} and the allele frequency threshold to call a consensus indel. The \ac{LSDV} test samples which were used to assess the workflow quality yield considerable results and complete consensus sequences for each sample. \\
\todo{add results/discussion about lsdv/capv experiment. which ref. to choose -> which clade, pre-workflow with vapor for sub-sequence of meaningful positions}
% \todoit We showed that for Capripoxviruses, by choosing a different reference sequence from another \ac{CaPV} member for mapping, the final consensus sequence does not diverge from the other consensus genomes built based on a reference sequence from the same species. Moreover, the low mutation rate of poxviruses and few variations among samples suggest that a fixed choice of curated reference sequence is judicious. \todo{true?} \\
Reconstructing the full-length genome from raw reads in the way the workflow is designed captures all genetic variation within the isolates and the information can be used for surveillance of outbreaks, vaccine development and tracing of the virus origins. The poxvirus workflow is published and available for download on WorkflowHub and Dockstore, two commonly used databases for versioned pipelines. Additional training material is linked and provided alongside the workflow, so that a user can be guided through the steps for a deeper understanding of the single workflow steps. 

\subsubsection*{Avian Influenza Virus Sequence Analysis}
The challenge in sequence analysis of the avian influenza virus lays in finding an ideal reference sequence for mapping. Assembling raw reads may result in a complete genome too, however is more computationally intensive respecting the size of the genome and may end in missambly regions where coverage criteria are not met. In the presented \ac{AIV} workflow, we use a new approach that compiles a hybrid reference from a large database that contains thousands of sequences per \ac{AIV} segment. The user of the workflow is not required to enter an arbitrarily selected reference genome, instead the workflow automatically finds the most similar sequence from the database by using the search classification tool \texttt{VAPOR}. While it works fast on a large number of input reads and a query database consisting of thousands of sequences, it selects the highest scoring sequence from the database per segment based on a scoring function on a weighted De Bruijn graph. This method is less biased than mapping raw reads to an assembled contig and yields reliable results with high identities between the query reads and the found sequence for each of the segments in all test samples. Alternative methods to find similar sequences to use as reference, such as read classification by looking up large databases of full influenza genomes are often complex, slow on the large number of reads and require expertise from the user which is not necessarily available. Reference-based mapping with an automatic search for a close reference allows analysis even of limited fragments while avoiding the user to judiciously choose a reference, and with regards to \textit{de novo} assembly performs better in genomic regions with not enough initial read data. Stacking the found reference segments to one reference sequence which is used for mapping, the workflow provides a fast method to build consensus sequence by avoiding \textit{de novo} assembly. The overall quality in the H4N6 and H5N8 test samples proves that this method finds the same consensus sequence as the assembly of the reads. These results encourage to test and use the workflow with more samples. The criteria for sequences within the reference genome database assure that complete proteins are captures within each gene by only retaining sequences that contain the complete start and stop codons and no ambiguous nucleotides. Moreover, the amount of sequences per subtype guarantees to capture within-subtype variation which is specifically vital for the detection and identification of nucleotide sites that differ from known strains. For investigation on base level on an \ac{AIV} genome, possible adaptations or reassortment events are captured and provide a useful starting point for further downstream analyses. The developed workflow provides multiple datapoints for examination within or outside Galaxy, and includes a summary of \acp{SNP} on per-segment level. Phylogenetic classification based on sequences from the reference database or extended with other samples outside the collection is a part of the workflow to allow insights into relations and clusters among the samples. Outbreak tracing to determine transmission events, intra-host variation and virus origins can be started from these data. Other possible analyses could include variant calling, lineage assignment, gene annotation, functional analysis and many more. These opportunities for downstream analysis are more exhaustive than other existing pipelines for genomic sequence analysis of \ac{AIV} samples, and while subtype identification remains an essential part of surveillance methods in the field, modern monitoring networks for zoonotic animal diseases require detailed study of the sequenced samples on the full genome. Although the workflow design is different to the underlying concept of the \ac{SARS-CoV-2} workflow, it picks up its major components while considering the \ac{AIV} specific viral attributes. 
Since this first version of the workflow ends by constructing the consensus sequence and marks some further directions for biomedical analysis, depending on research interests it could be expanded in the future.

\subsubsection*{Foot-and-mouth Disease Virus Sequence Analysis}
- fmdv wf: whole genome construction for such small virus genome with high genetic variation within one serotype and even "sub-type" makes it even more difficult to select reference sequence for mapping, short genome therefore tradeoff to use de novo assembly on the raw reads to construct a sequence to blast on, user is asked to pick result - still does not require much knowledge because of accompanying guidelines to be followed, wf is capable of detecting contamination or co-infection in the isolate. use blast result to download reference sequence for mapping in second workflow, reuse sars-cov-2 for proven to work consensus sequence construction. new approach in galaxy with low cost and time requirements, still serving base-by-base information for the sequenced sample. again phylogeny and other analyses are possible depending on the user. wf tested with four samples of different serotypes, consensus sequence was constructed and depends on number of raw reads for coverage, co-infection was detected, virus specific regions of polyc/polya tracts and termini are ambiguous and always pose challenge in assembly, still valuable insight to whole genome.

- summary: three new wgs workflows on galaxy, ready-to-use, embedded with additional resources, adaptable and transparent, allow for studying samples on base level, provide essential information for surveillance measures that are needed for modern virus containment
- highlight reference-based mapping vs. assembly
- contribute by open source concept on galaxy that is already known worldwide in bioinformatics, integrate into routine surveillance

%%% Interpret findings: Interpretation of results, explain what results mean and how they relate to research questions, describe any patterns, relationships in the data

%%% Discuss the implications of the findings: broader implications of research findings, explain how research contributes to the existing knowledge in the field and how it could impact future research, practice, or policy

\section{Limitations}
%%% Acknowledge limitations: discuss any potential sources of bias or confounding that could have affected the results, suggest ways to address these limitations in future research

\section{Future Directions}\label{sec:5-future}
