\chapter{Discussion}\label{chap:discussion}
% rephrase problems/questions
Despite enormous efforts in the prevention and containment of viral livestock diseases, viruses are a constant threat to the economy and the livestock industry. Extensive information on viruses is being collected at the international level, and research advances in diagnostics, epidemiology and virology are increasingly providing valuable insights into the genomic characteristics of specific viruses. Sequence analysis plays an important role in identifying outbreak sources and monitoring antigenic changes within known virus strains. Continuous surveillance, vaccine development and quality control are important aspects of global disease control programs to protect animals, livestock industry and public health. Therefore, a public infrastructure for modern surveillance methods for the use of \ac{NGS} is of global interest. Reconstruction of the full-length viral genome at the base level from raw sequencing reads requires practical knowledge in tool selection and parameter settings. However, \ac{WGS} is valuable for virus surveillance as it provides comprehensive genetic information about the virus sample. This information is used to identify virus strains and disease-causing mutations, detect within-host variation and determine relationships between populations and individual samples. Moreover, research on \ac{SARS-CoV-2} which was used to combat the current pandemic, has made such great strides in understanding viral structures that the introduction of a public infrastructure for experts and laypeople suggests that this newfound knowledge can be applied to other emerging viruses. Existing pipelines for full genomic sequence analysis from raw read data are specific to a small subset of viruses, require domain expertise and a server infrastructure to run computations on local computers. Different sequencing platforms produce different raw read data types with specific needs. As part of the \ac{ZODIAC} project, we have developed three ready-to-use, publicly available workflows on the Galaxy platform for sequence analysis of poxviruses, \ac{AIV} and \ac{FMDV} from Illumina-sequenced reads to analyse emerging viral diseases in livestock. With accompanying training resources and user-friendly workflows on an open-source platform, we ultimately contribute to research-based surveillance methods and enable health professionals without in-house resources to analyse viral outbreaks in livestock.

\section{Contribution to the Field}
% Summarise key findings: both the significant and non-significant results
%%%%%%%%%%
\todoit
- 3 workflows for livestock viruses of relevance, objective to construct whole genome by curative methods from NGS raw reads to consensus sequence, without required user knowledge that reuse concepts from sars-cov-2 workflows 
- illumina platform for sequencing
- poxvirus wf: new approach that requires sequencing in tiling amplicon manner, two pools, reference-based mapping, avoids assembly, reference sequence is ok to choose because mutation rate is low for poxviruses, consensus calling. challenge for poxviruses: two identical ITRs, solved by tiling amplicon approach. tested with LSDV samples, multisample workflow, analysis can be continued by subsequent analyses. WGS on lsdv isolates is essential to capture genetic variation all at once, in the past it was shown that outbreaks and vaccines are linked, careful monitoring required. workflow published on iwc/dockstore/workflowhub for versioning and distribution, additional training material that guides through the workflow step by step. both for beginners and experts to adapt and meaningful usage
- aiv wf: objective is to construct whole genome from NGS raw reads, challenge: find good reference sequence for mapping. new method for reference-based mapping that does not require the user to select a reference and imply knowledge about the virus strain in the isolate, or arbitrary selection; avoid de novo assembly, thus: hybrid reference that accounts for influenza gene-specific "best matching reference" on a high resolution level. use database of thousands of possible reference sequences and by vapor approach, select and stack reference for mapping. downstream components for consensus construction is similar to sars-cov-2 workflows and proves to be efficiently work for our two AIV test samples. for aiv, wgs is in high demand for threat of reassortment that causes adaption of the virus to mammals and spillover to humans, detection of mutation and study of key minor assets. with phylogenetic classification and lineage assignment gain insight on relations, geo/temporal clustering, tracing of single outbreaks and routes of transmission, virus origin. more detailed than "only" subtype identification, wgs in aiv surveillance is very crucial for worldwide monitoring. many other downstream options from the generated workflow outputs
- fmdv wf: whole genome construction for such small virus genome with high genetic variation within one serotype and even "sub-type" makes it even more difficult to select reference sequence for mapping, short genome therefore tradeoff to use de novo assembly on the raw reads to construct a sequence to blast on, user is asked to pick result - still does not require much knowledge because of accompanying guidelines to be followed, wf is capable of detecting contamination or co-infection in the isolate. use blast result to download reference sequence for mapping in second workflow, reuse sars-cov-2 for proven to work consensus sequence construction. new approach in galaxy with low cost and time requirements, still serving base-by-base information for the sequenced sample. again phylogeny and other analyses are possible depending on the user. wf tested with four samples of different serotypes, consensus sequence was constructed and depends on number of raw reads for coverage, co-infection was detected, virus specific regions of polyc/polya tracts and termini are ambiguous and always pose challenge in assembly, still valuable insight to whole genome.

- summary: three new wgs workflows on galaxy, ready-to-use, embedded with additional resources, adaptable and transparent, allow for studying samples on base level, provide essential information for surveillance measures that are needed for modern virus containment
- highlight reference-based mapping vs. assembly
- contribute by open source concept on galaxy that is already known worldwide in bioinformatics, integrate into routine surveillance

% Interpret findings: Interpretation of results, explain what results mean and how they relate to research questions, describe any patterns, relationships in the data

% Discuss the implications of the findings: broader implications of research findings, explain how research contributes to the existing knowledge in the field and how it could impact future research, practice, or policy

\section{Limitations}
% Acknowledge limitations: discuss any potential sources of bias or confounding that could have affected the results, suggest ways to address these limitations in future research

\section{Future Directions}
