%\todo{enlarge page margins to bottom}
% \newgeometry{bottom=0cm}
\begin{landscape}
	\ra{1.3}
	\begin{table}[b]
		\begin{picture}(0,0)
			\put(560,180){\rotatebox{-90}{\parbox{8cm}{\chapter*{Appendix}\label{chap:appendix} \section*{Poxvirus Workflow Outputs}}}}
		\end{picture}
		\begin{tabular}{|l|l|l|l|l|}
			\hline
			\textbf{No.} & \textbf{Output}                                                                                            & \textbf{From Tool}   & \textbf{Datatype} & \textbf{Remark}                                                                          \\ \hline \hline
			\textbf{12}  & \begin{tabular}[c]{@{}l@{}}Quality report of pool1/pool2 input\\ reads\end{tabular}                        & fastp                & HTML              & For web-view                                                                                \\ \hline
			\textbf{29}  & \begin{tabular}[c]{@{}l@{}}Alignment of pool1/pool2 against half-\\ masked reference sequence\end{tabular} & Map with \ac{BWA-MEM}     & \ac{BAM}               & \begin{tabular}[c]{@{}l@{}}Available as \acs{SAM} file in\\ hidden datasets\end{tabular}       \\ \hline
			\textbf{32}  & \begin{tabular}[c]{@{}l@{}}Statistics report of mapping of pool1/\\ pool2\end{tabular}                     & Samtools stats       & HTML              & For web-view                                                                                \\ \hline
			\textbf{31}  & \begin{tabular}[c]{@{}l@{}}Quality filtered alignments of pool1/\\ pool2\end{tabular}                      & Samtools view        & \ac{BAM}               & \begin{tabular}[c]{@{}l@{}}Available as \acs{SAM} file in\\ hidden datasets\end{tabular}       \\ \hline
			\textbf{39}  & Merged alignments                                                                                          & Samtools merge       & \ac{BAM}               & \begin{tabular}[c]{@{}l@{}}Available as \acs{SAM} file in\\ hidden datasets\end{tabular}       \\ \hline
			\textbf{40}  & \begin{tabular}[c]{@{}l@{}}Quality and coverage reports of\\ merged alignments\end{tabular}                & QualiMap BamQC       & HTML              & For web-view                                                                                \\ \hline
			\textbf{41}  & Primer-trimmed aligned reads                                                                               & iVar trim            & \ac{BAM}               & \begin{tabular}[c]{@{}l@{}}Available as \acs{SAM} file in\\ hidden datasets\end{tabular}       \\ \hline
			\textbf{43}  & Consensus sequences of each sample                                                                         & iVar consensus       & FASTA             & \begin{tabular}[c]{@{}l@{}}List of one FASTA file per\\ sample\end{tabular}              \\ \hline
			\textbf{47}  & Combined consensus genomes                                                                                 & Concatenate datasets & FASTA             & \begin{tabular}[c]{@{}l@{}}Consensus sequences of\\ all samples in one file\end{tabular} \\ \hline
			\end{tabular}
			\caption{Outputs of the Poxvirus workflow that can be used for downstream analyses.}
			\label{tab:pox-outputs}
	\end{table}
\end{landscape}

\begin{landscape}
	\ra{0.9}
		\begin{picture}(0,0)
			\put(580,10){\rotatebox{-90}{\parbox{8cm}{\section*{Poxvirus Workflow Steps}}}}
		\end{picture}
		\scriptsize
		% \tiny
			\begin{longtable}{|l|l|l|l|l|l|}
			\hline
			\textbf{\begin{tabular}[c]{@{}l@{}}Output\\ No.\end{tabular}} & \textbf{\begin{tabular}[c]{@{}l@{}}Input\\ No.\end{tabular}} & \textbf{Description}                                                                                                                          & \textbf{Tool}                                                                  & \textbf{Parameters}                                                                                                                                                                                                                                                                                                                           & \textbf{\begin{tabular}[c]{@{}l@{}}Output\\ Datatype\end{tabular}}           \\ \hline \hline \endhead
			\textbf{2}                                                     & -                                                             & \begin{tabular}[c]{@{}l@{}}Upload Primer scheme with\\ pool1/pool2 identifiers in SCORE\\ column\end{tabular}                                                   & -                                                                              & -                                                                                                                                                                                                                                                                                                                           & \ac{BED}                                                                          \\ \hline
			\textbf{9}                                                     & 2                                                             & Select pool1 primers                                                                                                                          & Select 1.0.4                                                                   & \begin{tabular}[c]{@{}l@{}}that: Matching\\ the pattern: {[}p|P{]}ool1\end{tabular}                                                                                                                                                                                                                                                           & \ac{BED}                                                                          \\ \hline
			\textbf{10}                                                    & 2                                                             & Select pool2 primers                                                                                                                          & Select 1.0.4                                                                   & \begin{tabular}[c]{@{}l@{}}that: Matching\\ the pattern: {[}p|P{]}ool2\end{tabular}                                                                                                                                                                                                                                                           & \ac{BED}                                                                          \\ \hline
			\textbf{14}                                                    & 9                                                             & Get end position of pool1                                                                                                                     & Datamash 1.1.0                                                                 & \begin{tabular}[c]{@{}l@{}}Operation to perform on each group:\\ Type: maximum\\ On column: 3\end{tabular}                                                                                                                                                                                                                                    & \ac{BED}                                                                          \\ \hline
			\textbf{15}                                                    & 10                                                            & Get start position of pool2                                                                                                                   & Datamash 1.1.0                                                                 & \begin{tabular}[c]{@{}l@{}}Operation to perform on each group:\\ Type: minimum\\ On column: 2\end{tabular}                                                                                                                                                                                                                                    & \ac{BED}                                                                          \\ \hline
			\textbf{19}                                                    & 14                                                            & Parse integer as text                                                                                                                         & \begin{tabular}[c]{@{}l@{}}Parse parameter value\\ 0.1.0\end{tabular}          & -                                                                                                                                                                                                                                                                                                                                             & text\_param                                                                  \\ \hline
			\textbf{20}                                                    & 15                                                            & Parse integer as text                                                                                                                         & \begin{tabular}[c]{@{}l@{}}Parse parameter value\\ 0.1.0\end{tabular}          & -                                                                                                                                                                                                                                                                                                                                             & text\_param                                                                  \\ \hline
			\textbf{1}                                                     & -                                                             & Upload reference sequence                                                                                                                     & -                                                                              & -                                                                                                                                                                                                                                                                                                                                             & FASTA                                                                        \\ \hline
			\textbf{8}                                                     & 1                                                             & Get the full length                                                                                                                           & \begin{tabular}[c]{@{}l@{}}Compute sequence\\ length 1.0.3\end{tabular}        & -                                                                                                                                                                                                                                                                                                                                             & tabular                                                                      \\ \hline
			\textbf{13}                                                    & 8                                                             & Get the length value from table                                                                                                               & Cut 1.0.2                                                                      & Cut columns: c2                                                                                                                                                                                                                                                                                                                               & tabular                                                                      \\ \hline
			\textbf{18}                                                    & 13                                                            & Parse integer as text                                                                                                                         & \begin{tabular}[c]{@{}l@{}}Parse parameter value\\ 0.1.0\end{tabular}          & -                                                                                                                                                                                                                                                                                                                                             & text\_param                                                                  \\ \hline
			\textbf{23}                                                    & 19, 18                                                        & \begin{tabular}[c]{@{}l@{}}Build interval for masking the\\ second half of the reference\end{tabular}                                         & \begin{tabular}[c]{@{}l@{}}Compose text\\ parameter value\\ 0.1.1\end{tabular} & \begin{tabular}[c]{@{}l@{}}1: components\\ value from 19\\ 2: components\\ Enter text that should be part of the\\ computed value: -\\ 3: components\\ value from 18\end{tabular}                                                                                                                                                             & text\_param                                                                  \\ \hline
			\textbf{24}                                                    & 20                                                            & \begin{tabular}[c]{@{}l@{}}Build interval for masking the\\ first half of the reference\end{tabular}                                          & \begin{tabular}[c]{@{}l@{}}Compose text\\ parameter value\\ 0.1.1\end{tabular} & \begin{tabular}[c]{@{}l@{}}1: components\\ Enter text that should be part of the\\ computed value: 1-\\ 2: components\\ value from 20\end{tabular}                                                                                                                                                                                            & text\_param                                                                  \\ \hline
			\textbf{27}                                                    & 1, 23                                                         & \begin{tabular}[c]{@{}l@{}}Mask reference for mapping of\\ pool1\end{tabular}                                                                 & maskseq 5.0.0                                                                  & Regions to mask: value from 23                                                                                                                                                                                                                                                                                                                & FASTA                                                                        \\ \hline
			\textbf{28}                                                    & 1, 24                                                         & \begin{tabular}[c]{@{}l@{}}Mask reference for mapping of\\ pool2\end{tabular}                                                                 & maskseq 5.0.0                                                                  & Regions to mask: value from 24                                                                                                                                                                                                                                                                                                                & FASTA                                                                        \\ \hline
			\textbf{3}                                                     & -                                                             & \begin{tabular}[c]{@{}l@{}}Upload paired-end reads of pool1\\ run\end{tabular}                                                                & -                                                                              & -                                                                                                                                                                                                                                                                                                                                             & \begin{tabular}[c]{@{}l@{}}list of fastq/\\ fastqsanger\\ pairs\end{tabular} \\ \hline
			\textbf{4}                                                     & -                                                             & \begin{tabular}[c]{@{}l@{}}Upload paired-end reads of pool2\\ run\end{tabular}                                                                & -                                                                              & -                                                                                                                                                                                                                                                                                                                                             & \begin{tabular}[c]{@{}l@{}}list of fastq/\\ fastqsanger\\ pairs\end{tabular} \\ \hline
			\textbf{11}                                                    & 3                                                             & Get the sample names from pool1                                                                                                               & \begin{tabular}[c]{@{}l@{}}Extract element\\ identifiers 0.0.2\end{tabular}    & -                                                                                                                                                                                                                                                                                                                                             & TXT                                                                          \\ \hline
			\textbf{17}                                                    & 4, 11                                                         & \begin{tabular}[c]{@{}l@{}}Sort samples of pool 2 according\\ to pool1\end{tabular}                                                           & Sort collection 1.0.0                                                          & Sort type: from file                                                                                                                                                                                                                                                                                                                          & \begin{tabular}[c]{@{}l@{}}list of fastq/\\ fastqsanger\\ pairs\end{tabular} \\ \hline
			\textbf{16}                                                    & 11                                                            & Place one sample identifier per file                                                                                                          & Split file 0.5.0                                                               & \begin{tabular}[c]{@{}l@{}}Select the file type to split: Text files\\ Base name for new files in collection:\\ split\_file\end{tabular}                                                                                                                                                                                                      & TXT                                                                          \\ \hline
			\textbf{21}                                                    & 16                                                            & Parse identifier as text                                                                                                                      & \begin{tabular}[c]{@{}l@{}}Parse parameter value\\ 0.1.0\end{tabular}          & -                                                                                                                                                                                                                                                                                                                                             & text\_param                                                                  \\ \hline
			\textbf{25}                                                    & 21                                                            & \begin{tabular}[c]{@{}l@{}}Append \_pool1 to the sample\\ identifiers\end{tabular}                                                            & \begin{tabular}[c]{@{}l@{}}Compose text\\ parameter value\\ 0.1.1\end{tabular} & \begin{tabular}[c]{@{}l@{}}1: components\\ value from 21\\ 2: components\\ Enter the text that should be part of \\ the computed value: \_pool1\end{tabular}                                                                                                                                                                                  & text\_param                                                                  \\ \hline
			\textbf{26}                                                    & 21                                                            & \begin{tabular}[c]{@{}l@{}}Append \_pool2 to the sample\\ identifiers\end{tabular}                                                            & \begin{tabular}[c]{@{}l@{}}Compose text\\ parameter value\\ 0.1.1\end{tabular} & \begin{tabular}[c]{@{}l@{}}1: components\\ value from 21\\ 2: components\\ Enter the text that should be part of \\ the computed value: \_pool2\end{tabular}                                                                                                                                                                                  & text\_param                                                                  \\ \hline
			\textbf{12}                                                    & 3                                                             & Preprocessing of pool1 reads                                                                                                                  & fastp 0.23.2                                                                   & \begin{tabular}[c]{@{}l@{}}Single-end or paired reads: Paired\\ collection\\ PolyG tail trimming: Automatic\\ trimming for Illumina NextSeq/\\ NovaSeq data\end{tabular}                                                                                                                                                                      & \begin{tabular}[c]{@{}l@{}}fastq/\\ fastqsanger,\\ HTML,\\ JSON\end{tabular} \\ \hline
			\textbf{22}                                                    & 17                                                            & Preprocessing of pool2 reads                                                                                                                  & fastp 0.23.2                                                                   & \begin{tabular}[c]{@{}l@{}}Single-end or paired reads: Paired\\ collection\\ PolyG tail trimming: Automatic\\ trimming for Illumina NextSeq/\\ NovaSeq data\end{tabular}                                                                                                                                                                      & \begin{tabular}[c]{@{}l@{}}fastq/\\ fastqsanger,\\ HTML,\\ JSON\end{tabular} \\ \hline
			\textbf{29}                                                    & 27, 12, 25                                                    & \begin{tabular}[c]{@{}l@{}}Mapping of pool1 against half-\\ masked reference sequence,\\ retaining read group identifier\end{tabular}         & \begin{tabular}[c]{@{}l@{}}Map with BWA-\\ MEM 0.7.17.2\end{tabular}           & \begin{tabular}[c]{@{}l@{}}Single or Paired-end reads: Paired\\ Collection\\ Set read groups information? Set read\\ groups (\acs{SAM}/\ac{BAM} specification)\\ Platform/technology used to produce\\ the reads (PL): ILLUMINA\end{tabular}                                                                                                             & \ac{BAM}                                                                          \\ \hline
			\textbf{30}                                                    & 28, 22, 26                                                    & \begin{tabular}[c]{@{}l@{}}Mapping of pool2 against half-\\ masked reference sequence,\\ retaining read group identifier\end{tabular}         & \begin{tabular}[c]{@{}l@{}}Map with BWA-\\ MEM 0.7.17.2\end{tabular}           & \begin{tabular}[c]{@{}l@{}}Single or Paired-end reads: Paired\\ Collection\\ Set read groups information? Set read\\ groups (\acs{SAM}/\ac{BAM} specification)\\ Platform/technology used to produce\\ the reads (PL): ILLUMINA\end{tabular}                                                                                                             & \ac{BAM}                                                                          \\ \hline
			\textbf{32}                                                    & 29                                                            & Generate statistics of mapping                                                                                                                & Samtools stats 2.0.4                                                           & -                                                                                                                                                                                                                                                                                                                                             & tabular                                                                      \\ \hline
			\textbf{34}                                                    & 30                                                            & Generate statistics of mapping                                                                                                                & Samtools stats 2.0.4                                                           & -                                                                                                                                                                                                                                                                                                                                             & tabular                                                                      \\ \hline
			\textbf{35}                                                    & 12, 32                                                        & \begin{tabular}[c]{@{}l@{}}Aggregate quality reports of pool1\\ reads and mapping statistics\end{tabular}                                     & MultiQC 1.11                                                                   & \begin{tabular}[c]{@{}l@{}}1: Results\\ Which tool was used to generate logs?\\ fastp\\ 2: Results\\ Which tool was used to generate logs?\\ Samtools\\ Type of Samtools output? stats\end{tabular}                                                                                                                                           & HTML                                                                         \\ \hline
			\textbf{37}                                                    & 22, 34                                                        & \begin{tabular}[c]{@{}l@{}}Aggregate quality reports of pool1\\ reads and mapping statistics\end{tabular}                                     & MultiQC 1.11                                                                   & \begin{tabular}[c]{@{}l@{}}1: Results\\ Which tool was used to generate logs?\\ fastp\\ 2: Results\\ Which tool was used to generate logs?\\ Samtools\\ Type of Samtools output? stats\end{tabular}                                                                                                                                           & HTML                                                                         \\ \hline
			\textbf{31}                                                    & 29                                                            & \begin{tabular}[c]{@{}l@{}}Quality filter the mapped reads\\ of pool1\end{tabular}                                                            & Samtools view 1.15.1                                                           & \begin{tabular}[c]{@{}l@{}}What would you like to look at? A filtered/\\ subsampled selection of reads\\ Filter by quality: 20\\ Require that these flags are set: Read is paired,\\ Read is mapped in a proper pair\end{tabular}                                                                                                             & \ac{BAM}                                                                          \\ \hline
			\textbf{33}                                                    & 30                                                            & \begin{tabular}[c]{@{}l@{}}Quality filter the mapped reads\\ of pool2\end{tabular}                                                            & Samtools view 1.15.1                                                           & \begin{tabular}[c]{@{}l@{}}What would you like to look at? A filtered/\\ subsampled selection of reads\\ Filter by quality: 20\\ Require that these flags are set: Read is paired,\\ Read is mapped in a proper pair\end{tabular}                                                                                                             & \ac{BAM}                                                                          \\ \hline
			\textbf{36}                                                    & 31, 33                                                        & \begin{tabular}[c]{@{}l@{}}Create paired collection from both\\ pool mappings\end{tabular}                                                    & Zip collections 1.0.0                                                          & \begin{tabular}[c]{@{}l@{}}Input 1: value from 31\\ Input 2: value from 33\end{tabular}                                                                                                                                                                                                                                                       & \ac{BAM}                                                                          \\ \hline
			\textbf{38}                                                    & 36                                                            & \begin{tabular}[c]{@{}l@{}}Add rules to distinguish between\\ pool1 and pool2\end{tabular}                                                    & Apply rules 1.1.0                                                              & \begin{tabular}[c]{@{}l@{}}1. Add column for identifier0.\\ 2. Add column for identifier1.\\ 3. Set columns A and B as List Identifier(s)\end{tabular}                                                                                                                                                                                        & \ac{BAM}                                                                          \\ \hline
			\textbf{39}                                                    & 38                                                            & \begin{tabular}[c]{@{}l@{}}Merge the alignments of both\\ pools\end{tabular}                                                                  & \begin{tabular}[c]{@{}l@{}}Samtools merge\\ 1.15.1\end{tabular}                & Alignments in \ac{BAM} format: value from 38                                                                                                                                                                                                                                                                                                       & \ac{BAM}                                                                          \\ \hline
			\textbf{40}                                                    & 39                                                            & \begin{tabular}[c]{@{}l@{}}Generate quality and coverage \\ report of mapping to check for the\\ middle part of the merged pools\end{tabular} & \begin{tabular}[c]{@{}l@{}}QualiMap BamQC\\ 2.2.2d\end{tabular}                & -                                                                                                                                                                                                                                                                                                                                             & \begin{tabular}[c]{@{}l@{}}fastq/\\ fastqsanger,\\ HTML\end{tabular}         \\ \hline
			\textbf{42}                                                    & 40                                                            & \begin{tabular}[c]{@{}l@{}}Remove failed reads from the\\ dataset\end{tabular}                                                                & \begin{tabular}[c]{@{}l@{}}Filter failed datasets\\ 1.0.0\end{tabular}         & Input Collection: value from 40                                                                                                                                                                                                                                                                                                               & \begin{tabular}[c]{@{}l@{}}fastq/\\ fastqsanger,\\ HTML\end{tabular}         \\ \hline
			\textbf{44}                                                    & 42                                                            & \begin{tabular}[c]{@{}l@{}}Place the two datasets from nested\\ collection into a list of reports\end{tabular}                                & Flatten collection 1.0.0                                                       & \begin{tabular}[c]{@{}l@{}}Input Collection: value from 42\\ Join collection identifiers using: underscore\end{tabular}                                                                                                                                                                                                                       & \begin{tabular}[c]{@{}l@{}}fastq/\\ fastqsanger\end{tabular}                 \\ \hline
			\textbf{46}                                                    & 44                                                            & \begin{tabular}[c]{@{}l@{}}Aggregate results from the\\ reports\end{tabular}                                                                  & MultiQC 1.11                                                                   & \begin{tabular}[c]{@{}l@{}}1: Results\\ Which tool was used to generate logs?\\ Qualimap\end{tabular}                                                                                                                                                                                                                                         & HTML                                                                         \\ \hline
			\textbf{41}                                                    & 39, 2                                                         & \begin{tabular}[c]{@{}l@{}}Trim the aligned reads to remove\\ primers\end{tabular}                                                            & ivar trim 1.3.1                                                                & \begin{tabular}[c]{@{}l@{}}\ac{BED} file with primer sequences and positions:\\ value from 39\\ Filter reads based on amplicon info: Yes, drop\\reads that extend beyond amplicon boundaries\\ Include reads not ending in any primer binding\\ sites? Yes\end{tabular}                                                                           & \ac{BAM}                                                                          \\ \hline
			\textbf{5}                                                     & -                                                             & \begin{tabular}[c]{@{}l@{}}Enter minimum quality score to call\\ base\end{tabular}                                                            & -                                                                              & -                                                                                                                                                                                                                                                                                                                                             & integer                                                                      \\ \hline
			\textbf{6}                                                     & -                                                             & Enter allele frequency to call SNV                                                                                                            & -                                                                              & -                                                                                                                                                                                                                                                                                                                                             & float                                                                        \\ \hline
			\textbf{7}                                                     & -                                                             & Enter allele frequence to call indel                                                                                                          & -                                                                              & -                                                                                                                                                                                                                                                                                                                                             & float                                                                        \\ \hline
			\textbf{43}                                                    & 41                                                            & \begin{tabular}[c]{@{}l@{}}Consensus sequence construction\\ from trimmed and merged\\ alignments\end{tabular}                                & ivar consensus 1.3.2                                                           & \begin{tabular}[c]{@{}l@{}}Minimum quality score threshold to count base:\\ value from 5\\ Minimum frequency threshold: value from 6\\ Minimum indel frequency threshold: value \\from 7\\ Minimum depth to call consensus: 50\\ How to represent positions with coverage less\\ than the minimum depth threshold: Represent\\ as N\end{tabular} & FASTA                                                                        \\ \hline
			\textbf{45}                                                    & 43                                                            & \begin{tabular}[c]{@{}l@{}}Relabel consensus sequences per\\ sample\end{tabular}                                                              & \begin{tabular}[c]{@{}l@{}}Text transformation\\ 1.1.1\end{tabular}            & \begin{tabular}[c]{@{}l@{}}File to process: value from 43\\ SED Program: /\textasciicircum{}\textgreater{}/s/Consensus\_(.*)\\ \_threshold\_.*/\textbackslash{}1\end{tabular}                                                                                                                                                                    & FASTA                                                                        \\ \hline
			\textbf{47}                                                    & 45                                                            & \begin{tabular}[c]{@{}l@{}}Get combined consensus genomes\\ in a multifasta file\end{tabular}                                                 & \begin{tabular}[c]{@{}l@{}}Concatenate datasets\\ 0.1.1\end{tabular}           & Datasets to concatenate: value from 45                                                                                                                                                                                                                                                                                                        & FASTA                                                                        \\ \hline
		\caption{The full Poxvirus workflow with tools, parameters and input/output connections.}
		\end{longtable}
		\label{tab:pox-tools-steps}
\end{landscape}

\newpage
\begin{landscape}
	\ra{1.3}
	\begin{table}[]
		\begin{picture}(0,0)
			\put(580,180){\rotatebox{-90}{\parbox{8cm}{\section*{\ac{AIV} Workflow Outputs}}}}
		\end{picture}
		\begin{tabular}{|l|l|l|l|l|}
			\hline
			\textbf{No.} & \textbf{Output}                                                                                          & \textbf{From Tool}  & \textbf{Datatype} & \textbf{Remark}                          \\ \hline \hline
			\textbf{6}   & Quality report of input reads                                                                            & fastp               & HTML              & In report                                \\ \hline
			\textbf{8}   & Scores of closest references                                                                             & VAPOR               & tabular           & One table per segment                    \\ \hline
			\textbf{24}  & \begin{tabular}[c]{@{}l@{}}Scores of sequences chosen for\\ hybrid reference sequence\end{tabular}       & VAPOR               & tabular           & Overview of scores                       \\ \hline
			\textbf{25}  & \begin{tabular}[c]{@{}l@{}}Alignment of reads against\\ hybrid reference sequence\end{tabular}           & \begin{tabular}[c]{@{}l@{}}Map with BWA-\\MEM\end{tabular}    & \ac{BAM}               & \begin{tabular}[c]{@{}l@{}}Available as \acs{SAM} file\\ in hidden datasets\end{tabular} \\ \hline
			\textbf{27}  & Quality filtered alignments                                                                              & Samtools view       & \ac{BAM}               & \begin{tabular}[c]{@{}l@{}}Available as \acs{SAM} file\\ in hidden datasets\end{tabular} \\ \hline
			\textbf{31}  & \begin{tabular}[c]{@{}l@{}}Quality and coverage report\\ on alignment\end{tabular}                       & QualiMap BamQC      & HTML              & In report                                \\ \hline
			\textbf{36}  & Consensus sequence                                                                                       & iVar consensus      & FASTA             & One FASTA file per segment               \\ \hline
			\textbf{41}  & \begin{tabular}[c]{@{}l@{}}Consensus sequences with\\ segment names as sequence identifiers\end{tabular} & Collapse Collection & FASTA             & Contains eight sequences                 \\ \hline
			\textbf{42}  & Plots to visualise SNPs                                                                                  & snipit              & PNG               & \ac{HA} and \ac{NA} plots in report                \\ \hline
			\textbf{44}  & Genome annotation file                                                                                   & Prokka              & FAA               & In report                                \\ \hline
			\textbf{43}  & Phylogenetic trees                                                                                       & IQ-Tree             & iqtree            & \ac{HA} and \ac{NA} trees in report                \\ \hline
		\end{tabular}
		\caption{Outputs of the Avian Influenza Virus workflow that can be used for downstream analyses.}
		\label{tab:aiv-outputs}
	\end{table}
\end{landscape}

\newpage
\begin{landscape}
	\ra{0.9}
		\begin{picture}(0,0)
			\put(580,10){\rotatebox{-90}{\parbox{8cm}{\section*{\ac{AIV} Workflow Steps}}}}
		\end{picture}
		\tiny
		\begin{longtable}{|l|l|l|l|l|l|}
		\hline
			\textbf{\begin{tabular}[c]{@{}l@{}}Output\\ No.\end{tabular}}  & \textbf{\begin{tabular}[c]{@{}l@{}}Input\\ No.\end{tabular}}  & \textbf{Description}                                                                                                                        & \textbf{Tool}                                                       & \textbf{Parameters}                                                                                                                                                                                                                                                                                                                                           & \textbf{\begin{tabular}[c]{@{}l@{}}Output\\Datatype\end{tabular}}                   \\ \hline \hline \endhead
			\textbf{1}                                                     & -                                                             & \begin{tabular}[c]{@{}l@{}}Upload reference sequence database\\ per segment\end{tabular}                                                    & -                                                                   & -                                                                                                                                                                                                                                                                                                                                                             & list                                                                                \\ \hline
			\textbf{2}                                                     & -                                                             & Upload paired-end Illumina reads                                                                                                            & -                                                                   & -                                                                                                                                                                                                                                                                                                                                                             & \begin{tabular}[c]{@{}l@{}}pair of fastq/\\ fastqsanger\end{tabular}                \\ \hline
			\textbf{6}                                                     & 2                                                             & Preprocessing                                                                                                                               & fastp 0.20.1                                                        & \begin{tabular}[c]{@{}l@{}}Single-end or paired reads: Paired Collection\\ Length required: 30\\ PolyG tail trimming: Automatic trimming for\\ IlluminaNextSeq/NovaSeq data\\ Cut by quality in front (5'): Yes \\ Cut by quality in tail (3'): Yes\\ Cutting mean quality: 30\end{tabular}                                                                   & \begin{tabular}[c]{@{}l@{}}pair of fastq/\\ fastqsanger,\\ HTML\\ JSON\end{tabular} \\ \hline
			\textbf{8}                                                     & 1, 6                                                          & Find closest reference per segment                                                                                                          & VAPOR 1.0.2                                                         & \begin{tabular}[c]{@{}l@{}}Type of sequencing data: Paired-end as\\ collection\\ Desired output: Return scores of best matches\\ Limit number of reported matches to: 0\\ Kmer Length: 21\\ Read kmer filtering threshold: 0.1\\ Coverage threshold for k-mer culling: 5\\ Minimum k-mer proportion: 0.0\\ Fraction of best seeds to extend: 1.0\end{tabular} & tabular                                                                             \\ \hline
			\textbf{9}                                                     & 8                                                             & Get sequence identifier                                                                                                                     & Replace 1.1.4                                                       & \begin{tabular}[c]{@{}l@{}}Find pattern: \textasciicircum{}.+\textbackslash{}t\textgreater{}()\$\\ Replace with: \$1\\ Find-Pattern is a regular expression: Yes\\ Replace all occurences of the pattern: Yes\\ Find and Replace text in: entire line\end{tabular}                                                                                            & tabular                                                                             \\ \hline
			\textbf{12}                                                    & 9                                                             & Get the first identifier of each segment                                                                                                    & Select first 1.0.2                                                  & Select first: 1                                                                                                                                                                                                                                                                                                                                               & tabular                                                                             \\ \hline
			\textbf{16}                                                    & 1, 12                                                         & \begin{tabular}[c]{@{}l@{}}Extract sequences from the database\\ according to the identifiers\end{tabular}                                  & seqtk\_subseq 1.3.1                                                 & \begin{tabular}[c]{@{}l@{}}Select source of sequence choices: FASTA/Q\\ ID list\end{tabular}                                                                                                                                                                                                                                                                  & FASTA                                                                               \\ \hline
			\textbf{20}                                                    & 16                                                            & \begin{tabular}[c]{@{}l@{}}Place all closest reference sequences\\ into one FASTA file\end{tabular}                                         & \begin{tabular}[c]{@{}l@{}}Collapse Collection\\ 5.1.0\end{tabular} & -                                                                                                                                                                                                                                                                                                                                                             & FASTA                                                                               \\ \hline
			\textbf{23}                                                    & 20                                                            & Shorten the identifiers                                                                                                                     & Replace 1.1.4                                                       & \begin{tabular}[c]{@{}l@{}}Find pattern: \textgreater{}({[}\textasciicircum{}|{]}+).+\$\\ Replace with: >\$1\\ Find-Pattern is a regular expression: Yes\\ Replace all occurences of the pattern: Yes\\ Find and Replace text in: entire line\end{tabular}                                                                                                    & FASTA                                                                               \\ \hline
			\textbf{25}                                                    & 6, 23                                                         & \begin{tabular}[c]{@{}l@{}}Mapping against hybrid reference\\ sequence\end{tabular}                                                         & \begin{tabular}[c]{@{}l@{}}Map with BWA-\\MEM 0.7.17.2\end{tabular} & \begin{tabular}[c]{@{}l@{}}Single or Paired-end reads: Paired Collection\\ Select analysis mode: 1.Simple Illumina mode\\ \ac{BAM} sorting mode: Sort by chromosomal\\ coordinates\end{tabular}                                                                                                                                                                    & \ac{BAM}                                                                                 \\ \hline
			\textbf{10}                                                    & 8                                                             & \begin{tabular}[c]{@{}l@{}}Pick columns with sequence identifier\\ and score\end{tabular}                                                   & Cut 1.0.2                                                           & Cut columns: c6,c1                                                                                                                                                                                                                                                                                                                                            & tabular                                                                             \\ \hline
			\textbf{14}                                                    & 10                                                            & Get the first identifier of each segment                                                                                                    & Select first 1.0.2                                                  & Select first: 1                                                                                                                                                                                                                                                                                                                                               & tabular                                                                             \\ \hline
			\textbf{18}                                                    & 14                                                            & \begin{tabular}[c]{@{}l@{}}Place all closest reference sequences in\\ one table\end{tabular}                                                & \begin{tabular}[c]{@{}l@{}}Collapse Collection\\ 5.1.0\end{tabular} & -                                                                                                                                                                                                                                                                                                                                                             & tabular                                                                             \\ \hline
			\textbf{3}                                                     & -                                                             & \begin{tabular}[c]{@{}l@{}}Generate a text file with names\\ ``segment 1'', ``segment 2'' etc.\end{tabular}                                 & \begin{tabular}[c]{@{}l@{}}Create text file\\ 1.1.0\end{tabular}    & \begin{tabular}[c]{@{}l@{}}1: selection\\ Characters to insert: segment 1\\ Specify the number of iterations by: User\\ defined number\\ How many times? 1\\ (repeat until segment 8)\end{tabular}                                                                                                                                                            & TXT                                                                                 \\ \hline
			\textbf{22}                                                    & 3, 18                                                         & \begin{tabular}[c]{@{}l@{}}Build table with segment name,\\ sequence identifier and score\end{tabular}                                      & Paste 1.0.0                                                         & -                                                                                                                                                                                                                                                                                                                                                             & tabular                                                                             \\ \hline
			\textbf{24}                                                    & 22                                                            & Reorder the columns                                                                                                                         & Cut 1.0.2                                                           & Cut columns: c3,c1,c2                                                                                                                                                                                                                                                                                                                                         & tabular                                                                             \\ \hline
			\textbf{27}                                                    & 25                                                            & Quality filter the mapped reads                                                                                                             & Samtools view 1.9                                                   & \begin{tabular}[c]{@{}l@{}}What would you like to look at? A filtered/\\ subsampled selection of reads\\ Filter by quality: 20\\ Require that these flags are set: Read is paired,\\ Read is mapped in a proper pair\end{tabular}                                                                                                                             & \ac{BAM}                                                                                 \\ \hline
			\textbf{29}                                                    & 27                                                            & \begin{tabular}[c]{@{}l@{}}Generate coverage and quality report\\ of alignment\end{tabular}                                                 & \begin{tabular}[c]{@{}l@{}}QualiMap BamQC\\ 2.2.2d\end{tabular}     & \begin{tabular}[c]{@{}l@{}}Skip duplicate reads: Unselect all\\ Number of bins to use in across-reference\\ plots: 40\end{tabular}                                                                                                                                                                                                                            & \acs{SAM}                                                                                 \\ \hline
			\textbf{31}                                                    & 29                                                            & Generate HTML report                                                                                                                        & MultiQC 1.9                                                         & \begin{tabular}[c]{@{}l@{}}Which tool was used to generate logs?\\ Qualimap (BamQC or RNASeq output)\end{tabular}                                                                                                                                                                                                                                             & HTML                                                                                \\ \hline
			\textbf{26}                                                    & 25                                                            & Convert header to \acs{SAM} format                                                                                                                & Samtools view 1.9                                                   & \begin{tabular}[c]{@{}l@{}}What would you like to look at? Just the\\ input header (-H)\\ What would you like to have reported?\\ The header in ... Output format: \acs{SAM}\end{tabular}                                                                                                                                                                           & \acs{SAM}                                                                                 \\ \hline
			\textbf{28}                                                    & 26                                                            & Get header lines starting with @SQ                                                                                                          & Select 1.0.4                                                        & \begin{tabular}[c]{@{}l@{}}that: Matching\\ the pattern: \textasciicircum{}@SQ.+\end{tabular}                                                                                                                                                                                                                                                                 & \acs{SAM}                                                                                 \\ \hline
			\textbf{30}                                                    & 28                                                            & \begin{tabular}[c]{@{}l@{}}Rewrite segment names and get subtype\\ information\\ from identifier\end{tabular}                               & Replace 1.1.4                                                       & \begin{tabular}[c]{@{}l@{}}Find pattern: \textasciicircum{}\textbackslash{}@SQ\textbackslash{}tSN:(.*)\textbackslash{}tLN:({[}0-9{]}+)\\ Replace with: \$1\t0\t\$2\\ Find-Pattern is a regular expression: Yes\\ Replace all occurences of the pattern: Yes\\ Find and Replace text in: entire line\end{tabular}                                              & \acs{SAM}                                                                                 \\ \hline
			\textbf{32}                                                    & 30                                                            & \begin{tabular}[c]{@{}l@{}}Place the alignment of one segment\\ in one file\end{tabular}                                                    & Split file 0.5.0                                                    & \begin{tabular}[c]{@{}l@{}}Select the file type to split: Text files\\ Specify number of output files or number\\ of records per file? Number of records per file\\ Chunk size: 1\\ Base name for new files in collection:\\ split\_file\\ Method to allocate records to new files:\\ Alternate output files\end{tabular}                                     & \acs{SAM}                                                                                 \\ \hline
			\textbf{33}                                                    & 30                                                            & Get columns with sequence identifiers                                                                                                       & Cut 1.0.2                                                           & Cut columns: c1                                                                                                                                                                                                                                                                                                                                               & tabular                                                                             \\ \hline
			\textbf{34}                                                    & 27, 32                                                        & Filter by mapped intervals                                                                                                                  & Samtools view 1.9                                                   & \begin{tabular}[c]{@{}l@{}}What would you like to look at? A filtered/\\ subsampled selection of reads\\ Filter by regions: Regions from \ac{BED} file\end{tabular}                                                                                                                                                                                                & \ac{BAM}                                                                                 \\ \hline
			\textbf{35}                                                    & 34, 33                                                        & \begin{tabular}[c]{@{}l@{}}Rename sequence identifiers to\\ segment names\end{tabular}                                                      & \begin{tabular}[c]{@{}l@{}}Relabel identifiers\\ 1.0.0\end{tabular} & \begin{tabular}[c]{@{}l@{}}How should the new labels be specified?\\ Using lines in a simple text file.\end{tabular}                                                                                                                                                                                                                                          & \ac{BAM}                                                                                 \\ \hline
			\textbf{36}                                                    & 35                                                            & Consensus sequence construction                                                                                                             & ivar consensus 1.3.1                                                & \begin{tabular}[c]{@{}l@{}}Minimum quality score threshold to count\\ base: 20\\ Minimum frequency threshold: 0.7\\ Minimum depth to call consensus: 10\\ Exclude regions with smaller depth than the\\ minimum threshold: No\\ Use N instead of - for regions with less than\\ minimum coverage: Yes\end{tabular}                                            & FASTA                                                                               \\ \hline
			\textbf{38}                                                    & 36                                                            & Use readable sequence identifiers                                                                                                           & Replace 1.1.4                                                       & \begin{tabular}[c]{@{}l@{}}Find pattern: \textasciicircum{}\textgreater{}Consensus\_(.*)\_threshold\_.*\\ Replace with: \textgreater{}\$1\\ Find-Pattern is a regular expression: Yes\\ Replace all occurences of the pattern: Yes\\ Find and Replace text in: entire line\end{tabular}                                                                       & FASTA                                                                               \\ \hline
			\textbf{41}                                                    & 38                                                            & \begin{tabular}[c]{@{}l@{}}Create multifasta file containing the\\ eight consensus sequences\end{tabular}                                   & \begin{tabular}[c]{@{}l@{}}Collapse Collection\\ 5.1.0\end{tabular} & -                                                                                                                                                                                                                                                                                                                                                             & FASTA                                                                               \\ \hline
			\textbf{44}                                                    & 41                                                            & \begin{tabular}[c]{@{}l@{}}Create genome annotation file from\\ consensus sequence\end{tabular}                                             & Prokka 1.14.6                                                       & Kingdom: Viruses                                                                                                                                                                                                                                                                                                                                              & FAA                                                                                 \\ \hline
			\textbf{11}                                                    & 9                                                             & \begin{tabular}[c]{@{}l@{}}Get identifiers of the 10 best scores per\\ segment\end{tabular}                                                 & Select first 1.0.2                                                  & Select first: 10                                                                                                                                                                                                                                                                                                                                              & tabular                                                                             \\ \hline
			\textbf{15}                                                    & 1, 11                                                         & \begin{tabular}[c]{@{}l@{}}Retrieve sequences of the 10 best scores\\ per segment from reference database\end{tabular}                      & seqtk\_subseq 1.3.1                                                 & \begin{tabular}[c]{@{}l@{}}Select source of sequence choices: FASTA/Q\\ ID list\end{tabular}                                                                                                                                                                                                                                                                  & tabular                                                                             \\ \hline
			\textbf{19}                                                    & 15                                                            & \begin{tabular}[c]{@{}l@{}}Multiple sequence alignment of the 10\\ best scores per segment\end{tabular}                                     & \acs{MAFFT} 7.508                                                         & \begin{tabular}[c]{@{}l@{}}Data type: Nucleic acids\\ Matrix selection: No matrix\end{tabular}                                                                                                                                                                                                                                                                & FASTA                                                                               \\ \hline
			\textbf{4}                                                     & -                                                             & Set size of phylogenetic trees                                                                                                              & -                                                                   & -                                                                                                                                                                                                                                                                                                                                                             & integer                                                                             \\ \hline
			\textbf{13}                                                    & 4, 9                                                          & \begin{tabular}[c]{@{}l@{}}Select the first X lines from the VAPOR\\ scores table\end{tabular}                                              & Select first 1.0.2                                                  & -                                                                                                                                                                                                                                                                                                                                                             & tabular                                                                             \\ \hline
			\textbf{5}                                                     & 1                                                             & \begin{tabular}[c]{@{}l@{}}Dismiss B strain if present in the\\ reference sequence database\end{tabular}                                    & Filter FASTA 2.3                                                    & \begin{tabular}[c]{@{}l@{}}Criteria for filtering on the headers: Regular\\ expression on the headers\\ Regular expression pattern the header\\ should match: \textbackslash{}\textgreater{}\textbackslash{}S*\textbackslash{}|A\end{tabular}                                                                                                                 & FASTA                                                                               \\ \hline
			\textbf{7}                                                     & 5                                                             & \begin{tabular}[c]{@{}l@{}}Dismiss H17 and H18 subtypes if\\ present in the reference sequence\\ database\end{tabular}                      & Filter FASTA 2.3                                                    & \begin{tabular}[c]{@{}l@{}}Criteria for filtering on the headers: Regular\\ expression on the headers\\ Regular expression pattern the header should\\ match:\\ H1N|H2|H3|H4|H5|H6|H7|H8|H9|H10|\\ H11|H12|H13|H14|H15|H16\end{tabular}                                                                                                                       & FASTA                                                                               \\ \hline
			\textbf{17}                                                    & 7, 13                                                         & \begin{tabular}[c]{@{}l@{}}Retrieve the sequences for the\\ phylogenetic trees from the filtered\\ reference sequence database\end{tabular} & seqtk\_subseq 1.3.1                                                 & \begin{tabular}[c]{@{}l@{}}Select source of sequence choices: FASTA/Q\\ ID list\end{tabular}                                                                                                                                                                                                                                                                  & FASTA                                                                               \\ \hline
			\textbf{21}                                                    & 17                                                            & \begin{tabular}[c]{@{}l@{}}Multiple sequence alignment of the\\ sequences for phylogentic trees\end{tabular}                                & \acs{MAFFT} 7.508                                                         & \begin{tabular}[c]{@{}l@{}}Data type: Nucleic acids\\ Matrix selection: No matrix\end{tabular}                                                                                                                                                                                                                                                                & FASTA                                                                               \\ \hline
			\textbf{37}                                                    & 36                                                            & \begin{tabular}[c]{@{}l@{}}Rename sequence identifiers from\\ consensus sequences\end{tabular}                                              & Replace 1.1.4                                                       & \begin{tabular}[c]{@{}l@{}}Find pattern: \textasciicircum{}\textgreater{}.+\\ Replace with: \textgreater{}sequenced\_sample\\ Find-Pattern is a regular expression: Yes\\ Replace all occurences of the pattern: Yes\\ Find and Replace text in: entire line\end{tabular}                                                                                     & FASTA                                                                               \\ \hline
			\textbf{39}                                                    & 37, 19                                                        & \begin{tabular}[c]{@{}l@{}}Add consensus sequences to the\\ alignment\end{tabular}                                                          & \acs{MAFFT} add 7.508                                                     & \begin{tabular}[c]{@{}l@{}}What do you want to add to the alignment: A\\ single sequence\\ Preserve the original alignment: Yes\\ Preserve the original order of sequences: Yes\end{tabular}                                                                                                                                                                  & FASTA                                                                               \\ \hline
			\textbf{42}                                                    & 39                                                            & \begin{tabular}[c]{@{}l@{}}Summarise SNPs relative to the\\ reference sequence of each segment\end{tabular}                                 & snipit 1.0.7                                                        & \begin{tabular}[c]{@{}l@{}}The reference sequence ...: should be picked\\ via its ID\\ ID of reference sequence: sequenced\_sample\\ Order of samples in the plot: Sort by number\\ of mutations\\ Invert sort order: Yes\\ Flip plot orientation: Yes\end{tabular}                                                                                           & \begin{tabular}[c]{@{}l@{}}Collection of\\ PNG\end{tabular}                         \\ \hline
			\textbf{45}                                                    & 42                                                            & Get \acs{SNP} plot for \ac{HA} segment                                                                                                                 & Extract dataset 1.0.1                                               & \begin{tabular}[c]{@{}l@{}}How should a dataset be selected? Select by\\ index\\ Element index: 3\end{tabular}                                                                                                                                                                                                                                                & PNG                                                                                 \\ \hline
			\textbf{46}                                                    & 42                                                            & Get \acs{SNP} plot for \ac{NA} segment                                                                                                                 & Extract dataset 1.0.1                                               & \begin{tabular}[c]{@{}l@{}}How should a dataset be selected? Select by\\ index\\ Element index: 5\end{tabular}                                                                                                                                                                                                                                                & PNG                                                                                 \\ \hline
			\textbf{40}                                                    & 38, 21                                                        & \begin{tabular}[c]{@{}l@{}}Add relabeled consensus sequences\\ to the alignment\end{tabular}                                                & \acs{MAFFT} add 7.508                                                     & \begin{tabular}[c]{@{}l@{}}What do you want to add to the alignment:\\ A single sequence\\ Preserve the original alignment: Yes\end{tabular}                                                                                                                                                                                                                  & FASTA                                                                               \\ \hline
			\textbf{43}                                                    & 40                                                            & \begin{tabular}[c]{@{}l@{}}Build phylogenetic trees for each\\ segment from the specific MSAs\end{tabular}                                  & IQ-TREE 2.1.2                                                       & Specify sequence type: \ac{DNA}                                                                                                                                                                                                                                                                                                                                    & \begin{tabular}[c]{@{}l@{}}nhx, mldist,\\ iqtree\end{tabular}                       \\ \hline
			\textbf{47}                                                    & 43                                                            & Get phylogenetic tree for \ac{HA} segment                                                                                                        & Extract dataset 1.0.1                                               & \begin{tabular}[c]{@{}l@{}}How should a dataset be selected? Select by\\ index\\ Element index: 3\end{tabular}                                                                                                                                                                                                                                                & iqtree                                                                              \\ \hline
			\textbf{48}                                                    & 43                                                            & Get phylogenetic tree for \ac{NA} segment                                                                                                        & Extract dataset 1.0.1                                               & \begin{tabular}[c]{@{}l@{}}How should a dataset be selected? Select by\\ index\\ Element index: 5\end{tabular}                                                                                                                                                                                                                                                & iqtree                                                                              \\ \hline
		\caption{The full \ac{AIV} workflow with tools, parameters and input/output connections.}
		\end{longtable}
		\label{tab:aiv-tools-steps}
\end{landscape}
\newpage

\begin{landscape}
\ra{1.3}
\begin{table}[]
	\begin{picture}(0,0)
		\put(580,180){\rotatebox{-90}{\parbox{8cm}{\section*{\ac{FMDV} Workflow Outputs}}}}
	\end{picture}
	\begin{tabular}{|l|l|l|l|l|}
	\hline
	\textbf{No.} & \textbf{Output}                                                                                                   & \textbf{From Tool}                                               & \textbf{Datatype} & \textbf{Remark}                                                                                       \\ \hline \hline
	4            & Quality report of preprocessed input reads                                                                        & fastp                                                            & HTML              & Trimmed and quality filtered                                                                          \\ \hline
	5            & Scores of closest references                                                                                      & VAPOR                                                            & tabular           &                                                                                                       \\ \hline
	13           & Alignment                                                                                                         & \begin{tabular}[c]{@{}l@{}}Map with BWA-\\ MEM\end{tabular}      & BAM               &                                                                                                       \\ \hline
	24           & Quality filtered alignment                                                                                        & \begin{tabular}[c]{@{}l@{}}Samtools view/\\ MultiQC\end{tabular} & HTML              & \begin{tabular}[c]{@{}l@{}}Failed datasets are filtered and\\ not included in the report\end{tabular} \\ \hline
	18           & Consensus sequences                                                                                               & iVar consensus                                                   & FASTA             & One FASTA file per sample                                                                             \\ \hline
	21           & \begin{tabular}[c]{@{}l@{}}Search results of consensus sequence query\\ in NCBI nucleotides database\end{tabular} & \begin{tabular}[c]{@{}l@{}}NCBI BLAST+\\ blastn\end{tabular}     & tabular           &                                                                                                       \\ \hline
	23           & Plots to visualise SNPs                                                                                           & snipit                                                           & PNG               & One plot per sample                                                                                   \\ \hline
	\end{tabular}
	\caption{Outputs of the Foot-and-mouth disease virus workflow that can be used for downstream analyses.}
	\label{tab:fmdv-outputs}
\end{table}
\end{landscape}

\newpage
\todoit{for FMDV, not AIV}
\begin{landscape}
	\ra{0.9}
	\begin{picture}(0,0)
		\put(580,10){\rotatebox{-90}{\parbox{8cm}{\section*{\ac{FMDV} Workflow Steps}}}}
	\end{picture}
	\tiny
	\begin{longtable}{|l|l|l|l|l|l|}
		\hline
		\textbf{\begin{tabular}[c]{@{}l@{}}Output\\ No.\end{tabular}}  & \textbf{\begin{tabular}[c]{@{}l@{}}Input\\ No.\end{tabular}}  & \textbf{Description}                                                                                                                        & \textbf{Tool}                                                       & \textbf{Parameters}                                                                                                                                                                                                                                                                                                                                           & \textbf{\begin{tabular}[c]{@{}l@{}}Output\\Datatype\end{tabular}}                   \\ \hline \hline \endhead
		
		\caption{The full \ac{FMDV} workflow with tools, parameters and input/output connections.}
	\end{longtable}
	\label{tab:fmdv-tools-steps}
\end{landscape}